src/comparator/service_comparator.py

```diff
@@ -119,6 +119,46 @@ def _compareRpcMethods(
                 FindingContainer.addFinding(
                     FindingCategory.METHOD_PAGINATED_RESPONSE_CHANGE, "", msg, True
                 )
+            # 6.8 The method_signature annotation is changed.
+            signatures_original = self._get_signatures(method_original)
+            signatures_update = self._get_signatures(method_update)
+            self._compare_method_signatures(signatures_original, signatures_update)
+
+    def _get_signatures(self, method: MethodDescriptorProto):
+        """Return the signature defined for this method."""
+        return method.options.Extensions[client_pb2.method_signature]
+
+    def _compare_method_signatures(self, signatures_original, signatures_update):
+        def _filter_fields(signatures: [str]) -> [str]:
+            fields = []
+            for sig in signatures:
+                for f in sig.split(','):
+                    if not f:
+                        # Special case for an empty signature
+                        continue
+                    name = f.strip()
+                    fields.append(name)
+            return fields
+        # Flatten the method_signatures fields. 
+        # For example: ['content, error'] to ['content', 'error']
+        fields_original = _filter_fields(signatures_original)
+        fields_update = _filter_fields(signatures_update)
+        
+        for num, f in enumerate(fields_original):
+            if num >= len(fields_update):
+                FindingContainer.addFinding(
+                    FindingCategory.METHOD_SIGNATURE_CHANGE,
+                    "",
+                    f"The existing method_signature {f} is removed.",
+                    True,
+                )
+            if fields_update[num] != f:
```

Comment:
I think this will cause an unexpected error if the update has fewer signatures than the original.

Reply:
If the num is out of index for fields_update, that would be caught by the first if-condition.

Reply:
Ah, I got the point. Changed the logic to the suggested one, thanks!